samples/nrf9160/nrf_cloud_rest_device_message/src/main.c

```diff
@@ -0,0 +1,429 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <modem/nrf_modem_lib.h>
+#include <nrf_modem_at.h>
+#include <modem/modem_info.h>
+#include <settings/settings.h>
+#include <net/nrf_cloud.h>
+#include <net/nrf_cloud_rest.h>
+#include <sys/reboot.h>
+#include <logging/log.h>
+#include <dk_buttons_and_leds.h>
+
+LOG_MODULE_REGISTER(nrf_cloud_rest_device_message,
+				    CONFIG_NRF_CLOUD_REST_DEVICE_MESSAGE_SAMPLE_LOG_LEVEL);
+
+#define BTN_NUM			CONFIG_REST_DEVICE_MESSAGE_BUTTON_EVT_NUM
+#define LED_NUM			CONFIG_REST_DEVICE_MESSAGE_LED_NUM
+#define JITP_REQ_WAIT_SEC	10
+
+/* Semaphore to indicate a button has been pressed */
+static K_SEM_DEFINE(button_press_sem, 0, 1);
+
+/* Semaphore to indicate that a network connection has been established */
+static K_SEM_DEFINE(lte_connected, 0, 1);
+
+/* nRF Cloud device ID */
+static char device_id[NRF_CLOUD_CLIENT_ID_MAX_LEN + 1];
+
+#define REST_RX_BUF_SZ 2100
+/* Buffer used for REST calls */
+static char rx_buf[REST_RX_BUF_SZ];
+
+#define JWT_DURATION_S	(60*5)
+#define JWT_BUF_SZ	900
+/* Buffer used for JSON Web Tokens (JWTs) */
+static char jwt[JWT_BUF_SZ];
+
+/* nRF Cloud REST context */
+struct nrf_cloud_rest_context rest_ctx = {
+	.connect_socket = -1,
+	.keep_alive = true,
```

Comment:
this should probably be set to false.
nrf cloud will close an idle socket in ~60s.

alternatively, handle a return value of `-ENOTCONN` from `nrf_cloud_rest_send_device_message`... which would involve calling `(void)nrf_cloud_rest_disconnect(&rest_ctx);` and then calling `nrf_cloud_rest_send_device_message` again.

Reply:
Ah, got it, thank you! I think that should be made a little clearer in documentation. I will make a note to add some clarifications there.

For simplicity's sake, I think I'd rather set keep-alive to false.